### PR TITLE
uefi: Allow overriding ESP mount point via conf file (Fixes: #421)

### PR DESCRIPTION
**Pull request by superm1**

### CODE REVIEW

File: plugins/uefi/fu-plugin-uefi.c [hunk 1]
```diff
@@ -43,11 +44,19 @@ fu_plugin_init (FuPlugin *plugin)
...
 	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_RUN_AFTER, "upower");
 	fu_plugin_add_report_metadata (plugin, "FwupdateVersion", LIBFWUP_LIBRARY_VERSION);
 	fu_plugin_add_report_metadata (plugin, "EfivarVersion", EFIVAR_LIBRARY_VERSION);
 }
 
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+        FuPluginData *data = fu_plugin_get_data (plugin);
+        g_free(data->esp_path);
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: Space after the bracket pls.

File: plugins/uefi/fu-plugin-uefi.c [hunk 2]
```diff
@@ -388,6 +397,10 @@ fu_plugin_update (FuPlugin *plugin,
 	g_debug ("Performing UEFI capsule update");
 	fu_device_set_status (device, FWUPD_STATUS_SCHEDULING);
 
+#ifdef HAVE_FWUP_CUSTOM_ESP
+	if (data->esp_path)
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: != NULL pls

File: plugins/uefi/fu-plugin-uefi.c [hunk 3]
```diff
@@ -582,6 +595,41 @@ fu_plugin_uefi_test_secure_boot (FuPlugin *plugin)
 	fu_plugin_add_report_metadata (plugin, "SecureBoot", result_str);
 }
 
+static gboolean load_config_file (FuPlugin *plugin, GError **error)
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: I'm not super keen on this. I'd be okay with something abstract like `gchar *fu_plugin_get_setting(plugin, const gchar *key)` that automatically loads the right config file -- and then it stops each plugin having a different version of the same thing. The key file system works now, but we might want a different config system in the future, hence why I'd like to abstract it out a little.

File: plugins/uefi/fu-plugin-uefi.c (hunk 1 above)
Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: Fixed

File: plugins/uefi/fu-plugin-uefi.c (hunk 2 above)
Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: Fixed

File: plugins/uefi/fu-plugin-uefi.c (hunk 3 above)
Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: Adjusted (and for your other comment that got lost in the force push)

File: plugins/uefi/fu-plugin-uefi.c [hunk 4]
```diff
@@ -582,6 +595,30 @@ fu_plugin_uefi_test_secure_boot (FuPlugin *plugin)
 	fu_plugin_add_report_metadata (plugin, "SecureBoot", result_str);
 }
 
+static gboolean load_custom_esp (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	const gchar *key = "OverrideESPMountPoint";
+
+	data->esp_path = fu_plugin_get_setting (plugin, key);
+	if (data->esp_path) {
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: != NULL

File: plugins/uefi/fu-plugin-uefi.c [hunk 5]
```diff
@@ -582,6 +595,30 @@ fu_plugin_uefi_test_secure_boot (FuPlugin *plugin)
...
+		if (!g_file_test (data->esp_path, G_FILE_TEST_IS_DIR)) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_FILE,
+				     "Invalid %s specified: %s",
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: Maybe also specify where the "invalid" value was set?

File: plugins/uefi/fu-plugin-uefi.c [hunk 6]
```diff
@@ -582,6 +595,30 @@ fu_plugin_uefi_test_secure_boot (FuPlugin *plugin)
...
+				     key, data->esp_path);
+
+			return FALSE;
+		}
+		g_debug("%s set to %s", key, data->esp_path);
```
Comment (hughsie) [inline plugins/uefi/fu-plugin-uefi.c]: space after bracket pls

**Comment by hughsie** [comment]
With those extra few nits, and assuming it passes CI, LGTM. Thanks.

File: plugins/uefi/fu-plugin-uefi.c (hunk 5 above)
Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: As in whole path?  or just "UEFI conf" file?  Sorta gets rid of abstraction if the plugin knows about the whole path here.

Comment (hughsie) [reply plugins/uefi/fu-plugin-uefi.c]: True, maybe leave that for now.

Comment (superm1) [reply plugins/uefi/fu-plugin-uefi.c]: I just changed it to refer to `"%s config", fu_plugin_get_name (plugin)` which will keep it abstracted but still point them where to fix it.

**Pull request merged**